pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -98,11 +99,20 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 		return errors.Wrapf(err, "failed to generate random string")
 	}
 
+	var zones *[]string
```

Comment:
can yo do this with var zones []string?

Reply:
seconded

Reply:
Problem with azure api, it doesnt like empty zone values, requires a nil to be passed instead of empty, passing empty causes it to error out :(

Reply:
We can still declare it as `var zones []string` and pass the pointer to `zones` as a parameter.

Reply:
so pointer to a nil slice is still not nil 

https://play.golang.org/p/zEGKmQn79eq 

if i try to pass in empty value, azure will error mentioning invalid zones.

Reply:
Does that matter though? The idiomatic way to check for a slice is by using the `len` method.

Reply:
So, at the time of retrieving the slice, I would just de reference the pointer and check its `len`.

Reply:
Yes. I dont want to add some complex code here

compute.VirtualMachine{
Zones: <>

We could change this to

vm := compute.VirtualMachine{}
if() {
vm.Zones = <>
}

if thats preferable, i feel the code would be the same though?